### PR TITLE
amd-gpu: Set the GPU name from libdrm

### PR DESCRIPTION
**Pull request by superm1**
libdrm has a table that will show the proper marketing name for GPUs as they get released.  Use this for device name.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/amd-gpu/fu-amd-gpu-device.c [hunk 1]
```diff
@@ -104,6 +104,31 @@ fu_amd_gpu_device_probe(FuDevice *device, GError **error)
...
+	r = amdgpu_device_initialize(fd, &major, &minor, &device_handle);
+	if (r) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "failed to initialize: %s",
+			    g_strerror(r));
+		return FALSE;
+	}
+	g_info("DRM %u.%u", major, minor);
```
Comment (hughsie) [inline plugins/amd-gpu/fu-amd-gpu-device.c:126]: Maybe store in ->priv and print in to_string?

**Review by hughsie — APPROVED** [review]

File: plugins/amd-gpu/fu-amd-gpu-device.c [hunk 2]
```diff
@@ -104,6 +115,30 @@ fu_amd_gpu_device_probe(FuDevice *device, GError **error)
...
+{
+	g_autoptr(FuDeviceLocker) locker = fu_device_locker_new(device, error);
+	FuIOChannel *io_channel = fu_udev_device_get_io_channel(FU_UDEV_DEVICE(device));
+	FuAmdGpuDevice *self = FU_AMDGPU_DEVICE(device);
+	amdgpu_device_handle device_handle;
+	gint fd, r;
+
+	fd = fu_io_channel_unix_get_fd(io_channel);
+	r = amdgpu_device_initialize(fd, &self->drm_major, &self->drm_minor, &device_handle);
+	if (r) {
```
Comment (hughsie) [inline plugins/amd-gpu/fu-amd-gpu-device.c:129]: `rc != 0` perhaps?

File: plugins/amd-gpu/fu-amd-gpu-device.c [hunk 3]
```diff
@@ -104,6 +115,30 @@ fu_amd_gpu_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_amd_gpu_device_set_marketing_name(FuDevice *device, GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = fu_device_locker_new(device, error);
+	FuIOChannel *io_channel = fu_udev_device_get_io_channel(FU_UDEV_DEVICE(device));
+	FuAmdGpuDevice *self = FU_AMDGPU_DEVICE(device);
+	amdgpu_device_handle device_handle;
```
Comment (hughsie) [inline plugins/amd-gpu/fu-amd-gpu-device.c:124]: can we do `= {0}` here?

Comment (superm1) [reply plugins/amd-gpu/fu-amd-gpu-device.c:124]: Should be fine to do, sure.

File: plugins/amd-gpu/fu-amd-gpu-device.c [hunk 4]
```diff
@@ -104,6 +115,32 @@ fu_amd_gpu_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_amd_gpu_device_set_marketing_name(FuDevice *device, GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = fu_device_locker_new(device, error);
```
Comment (hughsie) [inline plugins/amd-gpu/fu-amd-gpu-device.c:121]: I'm confused why this is needed -- the base device should already be open when setup() is called.

Comment (superm1) [reply plugins/amd-gpu/fu-amd-gpu-device.c]: Oh; you're right!  I fixed.  I also made it non-fatal if libdrm failed.  If it does we at least have the name from udev to go on.

**Review by hughsie — APPROVED** [review]

**Pull request merged**